File: algoexpert/python/medium/powerset.py
def powerset(arr):
    res = []
    temp = []

    def backtrack(i):
        if i >= len(arr):
            res.append(temp[::])
            return

        temp.append(arr[i])
        backtrack(i + 1)

        temp.pop()
        backtrack(i + 1)

    backtrack(0)
    return res

File: algoexpert/javascript/medium/powerset.js
function powerset(arr) {
  const res = [];
  let temp = [];

  function backtrack(i) {
    if (i >= arr.length) {
      res.push(temp.slice());
      return;
    }

    temp.push(arr[i]);
    backtrack(i + 1);

    temp.pop();
    backtrack(i + 1);
  }
  backtrack(0);
  return res;
}